SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Windows.Shared/Ingestion/Models/EventLog.cs

```diff
@@ -46,15 +41,24 @@ public partial class EventLog : LogWithProperties
         /// Concrete types like StartSessionLog or PageLog are always part of a
         /// session and always include this identifier.
         /// </param>
+        /// <param name="userId">optional string used for associating logs with
+        /// users.
+        /// </param>
         /// <param name="properties">Additional key/value pair parameters.
         /// </param>
-        public EventLog(System.DateTime? timestamp, Device device, System.Guid id, string name, System.Guid? sid = default(System.Guid?), IDictionary<string, string> properties = default(IDictionary<string, string>))
-            : base(device, timestamp, sid, properties)
+        public EventLog(Device device, System.Guid id, string name, System.DateTime? timestamp = default(System.DateTime?), System.Guid? sid = default(System.Guid?), string userId = default(string), IDictionary<string, string> properties = default(IDictionary<string, string>))
+            : base(device, timestamp, sid, userId, properties)
         {
             Id = id;
             Name = name;
+            CustomInit();
         }
 
+        /// <summary>
+        /// An initialization method that performs custom operations like setting defaults
```

Comment:
```suggestion
        /// An initialization method that performs custom operations like setting defaults.
```

Reply:
We don't want to change generated models. This should be changed in YAML.